src/main/java/online/shop/controller/admin/DeleteProductAdminController.java

```diff
@@ -0,0 +1,24 @@
+package online.shop.controller.admin;
+
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import online.shop.lib.Injector;
+import online.shop.service.ProductService;
+
+public class DeleteProductAdminController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("online.shop");
+    private final ProductService productService =
+            (ProductService) injector.getInstance(ProductService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String productId = req.getParameter("id");
+        Long id = Long.valueOf(productId);
+        productService.delete(id);
+        resp.sendRedirect(req.getContextPath() + "/admin/product");
```

Comment:
we should also do the same after adding a new product to the storage in the AddProductController
`resp.sendRedirect(req.getContextPath() + "/admin/product");`